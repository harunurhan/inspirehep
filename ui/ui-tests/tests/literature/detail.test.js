const { routes } = require('../../utils/constants');
const { createPollyInstance } = require('../../utils/polly');
const {
  takeScreenShotForDesktop,
  takeScreenShotForMobile,
} = require('../../utils/screenshot');

describe('Literature Detail', () => {
  let polly;

  beforeEach(async () => {
    await page.setRequestInterception(true);
    polly = createPollyInstance('LiteratureDetail');

    await page.goto(routes.public.literatureDetail1472986, {
      waitUntil: 'networkidle0',
    });
  });

  it('should match image snapshot for a literature', async () => {
    const desktopSS = await takeScreenShotForDesktop(page);
    // FIXME: citations by year graph is rendered incorrectly sometimes, therefore `failureThreshold`
    expect(desktopSS).toMatchImageSnapshot({
      failureThreshold: '0.04',
      failureThresholdType: 'percent',
    });

    const mobileSS = await takeScreenShotForMobile(page);
    expect(mobileSS).toMatchImageSnapshot();
  });

  it('sets literature title as document title', async () => {
    const documentTitle = await page.title();

    expect(documentTitle).toMatch(/^Estimating/);
  });

  afterEach(async () => {
    await polly.stop();
  });
});
