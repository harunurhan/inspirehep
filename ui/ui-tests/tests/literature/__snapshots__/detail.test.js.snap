// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Literature Detail sets abstract as meta description 1`] = `"The interference between Cabibbo-favored and Cabibbo-suppressed B→Dπ decay amplitudes provides sensitivity to the CKM angle γ. The relative size of the inter..."`;
